R/utils-formula.R

```diff
@@ -0,0 +1,58 @@
+#' Extract Formula Terms used for Covariance Structure Definition
+#'
+#' @param f (`formula`)\cr a formula from which covariance terms should be
+#'   extracted.
+#'
+#' @return A list of covariance structure expressions found in `f`.
+#'
+#' @importFrom stats terms
+#' @keywords internal
+extract_covariance_terms <- function(f) {
```

Comment:
a formula like `a ~ a+ ar1(a,b|c/d) ` will give error because the term in the left is also in the right part; using the `length(f)>` to decide the position is not robust here

Reply:
of cource `a ~ a + ar1(a,b|c,d)` should not be used; but this is a legal formula